fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -49,10 +56,27 @@ public FabricItemSettings equipmentSlot(EquipmentSlotProvider equipmentSlotProvi
 	 * @see CustomDamageHandler
 	 */
 	public FabricItemSettings customDamage(CustomDamageHandler handler) {
-		FabricItemInternals.computeExtraData(this).customDamage(handler);
+		this.customSettings.put(FabricItemInternals.CUSTOM_DAMAGE_HANDLER, handler);
+		return this;
+	}
+
+	/**
+	 * Sets a custom setting of the item.
+
+	 * @param setting the unique type for this setting
+	 * @param value the object containing the setting itself
+	 * @return this builder
+	 */
+	public <T> FabricItemSettings customSetting(CustomItemSetting<T> setting, T value) {
+		this.customSettings.put(setting, value);
 		return this;
 	}
 
+	@ApiStatus.Internal
+	public Map<CustomItemSetting<?>, ?> getCustomSettings() {
```

Comment:
This method is not very useful for actual api consumers.

Reply:
I believe that was the reasoning for external storage of the extra data?